crates/sui-config/src/node.rs

```diff
@@ -331,6 +331,14 @@ impl NodeConfig {
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum ConsensusProtocol {
```

Comment:
Node config is probably not the right place for the consensus protocol.
It's something that everyone must agree on, which usually is part of the protocol config, instead of node config.
Is this temporary?

Reply:
Yeah ultimately we want to have as part of protocol config, but I've explicitly avoided introducing a property there as the whole stitching at the moment is quite experimental. Yes that's temporary.

Reply:
One sad consequence though is people may start to have this field in their node config file, which is really not ideal.

Reply:
Will this field automatically get populated or people have to explicitly opt in? Maybe we can add an `experimental_` prefix. Or we can leave the logic in a commit ..

Reply:
It won't automatically get populated in the sense that the binary isn't able to modify the config file. But anytime anyone generates a fresh config (or when we refresh our config template) it will get included

Reply:
@lxfind I see the concern. What if then we just skip the ser/des for the `protocol` field as an additional protection? Right now we don't have a need to read from file.